lib/jsonapi/utils/response/formatters.rb

```diff
@@ -96,9 +106,18 @@ def build_response_document(object, options)
 
           if object.respond_to?(:to_ary)
             records = build_collection(object, options)
-            results.add_result(JSONAPI::ResourcesOperationResult.new(:ok, records, result_options(object, options)))
+
+            if options[:source_resource].present? && options[:relationship_type].present?
```

Comment:
That's a good idea. I wasn't aware the `params` object is available within this method.